aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -289,13 +281,14 @@ void avg_pool2d_out_cuda_template(
     inputHeight, inputWidth,
     outputHeight, outputWidth);
 
+  const auto memory_format = input_.suggest_memory_format();
   Tensor input = input_.contiguous(memory_format);
 
   output.resize_({nbatch, nInputPlane, outputHeight, outputWidth});
 
   const int32_t count = safe_downcast<int32_t, int64_t>(output.numel());
   const uint32_t num_threads = std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
-  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count, num_threads);
+  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count ? count : 1, num_threads);
```

Comment:
YOu don't need non-zero blocks here if you are going to skip kernel launch anyway.